catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java

```diff
@@ -100,9 +102,6 @@ public CreateTableRequest toCreateTableRequest() {
   String transformDistribution(Distribution distribution) {
     switch (distribution.strategy()) {
       case HASH:
-        Preconditions.checkArgument(
```

Comment:
with this , we will get the error.
```
Failed to operate table(s) [iceberg_bucket_partition_table] operation [CREATE] under schema [default_db], reason [Iceberg's Distribution Mode.HASH does not support set expressions.]
java.lang.IllegalArgumentException: Iceberg's Distribution Mode.HASH does not support set expressions.
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143)
	at com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergTable.transformDistribution(IcebergTable.java:105)
	at com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergTable.toCreateTableRequest(IcebergTable.java:83)
	at com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergCatalogOperations.createTable(IcebergCatalogOperations.java:512)
	at com.datastrato.gravitino.catalog.TableOperationDispatcher.lambda$createTable$7(TableOperationDispatcher.java:165)
	at com.datastrato.gravitino.catalog.CatalogManager$CatalogWrapper.lambda$doWithTableOps$1(CatalogManager.java:108)
	at com.datastrato.gravitino.utils.IsolatedClassLoader.withClassLoader(IsolatedClassLoader.java:72)
	at com.datastrato.gravitino.catalog.CatalogManager$CatalogWrapper.doWithTableOps(CatalogManager.java:103)
	at com.datastrato.gravitino.catalog.TableOperationDispatcher.lambda$createTable$8(TableOperationDispatcher.java:163)
	at com.datastrato.gravitino.catalog.OperationDispatcher.doWithCatalog(OperationDispatcher.java:89)
	at com.datastrato.gravitino.catalog.TableOperationDispatcher.createTable(TableOperationDispatcher.java:160)
	at com.datastrato.gravitino.catalog.TableEventDispatcher.createTable(TableEventDispatcher.java:99)
	at com.datastrato.gravitino.server.web.rest.TableOperations.lambda$createTable$2(TableOperations.java:106)
```

Reply:
this seems a bug?  `bucket(num, column)` is partition for Iceberg, not distribution

Reply:
I think it's reasonable to remove this limitation.

The bucket(num, column) is indeed the partition for iceberg and not the distribution, but the distribution here is converted from the transforms, [code](https://github.com/datastrato/gravitino/blob/main/spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/catalog/BaseCatalog.java#L169-L170).

The logic here is simply to get the distribution name and [set it to the write.distribution-mode parameter ](https://github.com/datastrato/gravitino/blob/main/catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java#L81)to distrbute the data before the write operation is performed.

Even if I create an iceberg bucket partition table by BucketTransform, I can actually change write.distribution-mode to none instead of hash after I create the table, just not adding a shuffle node before writing. However, the iceberg partition still exists, and the data will still be written into the correct partition path. Another difference is that hash distribution uses the iceberg partition function as the hash parameter, so that the data of the same partition can be gathered into one task for processing, which can reduce the number of small files.

Reply:
However, there are other limitations.   As long as the table is partitioned and sort-order is not specified, iceberg will set the distribution-mode to a hash, gravitino only set the distribution-mode to hash only when the transform is BucketTransform.  https://github.com/apache/iceberg/blob/228fc9b41f99f423d500a236c68216f0203dd42c/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java#L236-L272


But this can be optimized for other PR in the future.

Reply:
cc @FANNG1

Reply:
after [this](https://github.com/datastrato/gravitino/pull/2709#issuecomment-2063240320), we can roll back this change. WDYT? cc @FANNG1